ducktape/services/service.py

```diff
@@ -25,18 +25,29 @@
 # that there won't be resource conflicts: the cluster tests are being run on
 # should be large enough to use one instance per service instance.
 
-from ducktape.logger import Logger
 
-
-class Service(Logger):
-    def __init__(self, cluster, num_nodes):
-        self.num_nodes = num_nodes
-        self.cluster = cluster
+class Service(object):
+    def __init__(self, service_context):
+        """
+        :type service_context: ServiceContext
+        """
+        self.num_nodes = service_context.num_nodes
+        self.cluster = service_context.cluster
+        self.logger = service_context.logger
 
     def start(self):
         """Start the service running in the background."""
         self.nodes = self.cluster.request(self.num_nodes)
         for idx, node in enumerate(self.nodes, 1):
+
+            # Remote accounts utilities should log where this service logs
+            if node.account.logger is not None:
+                raise RuntimeError(
+                    "logger was not None on service start. There may be a concurrency issue, " +
+                    "or some service which isn't properly cleaning up after itself. " +
+                    "Service: %s, node.account: %s" % (self.__class__.__name__, str(node.account)))
```

Comment:
This tells you which Service ran into the problem, but doesn't necessarily help you track down the offending Service/Test. How about logging something to node.account.logger at critical and mentioning that in this message, e.g. say that you can grep the logs for 'MISBEHAVING TEST' or something so they know how to track down the problem?